src/main/java/com/boot/product/service/ProductService.java

```diff
@@ -118,6 +118,14 @@ public ProductDTO getProductBySlug(String slug, Boolean includeInactive) {
         return productEntityToDto(product);
     }
 
+    public List<String> getProductsByPartialName(String name) {
+
+        log.info("getProductsByPartialText - process started");
+
+        return productRepository.findByNameContainingIgnoreCase(name).stream().map(Product::getName)
+                .collect(Collectors.toList()).stream().limit(8).collect(Collectors.toList());
```

Comment:
this is where you limit the number of products. so basically you retrieve all matching products from db and limit the returned ones here

Reply:
yea, should i change something here?

Reply:
yes. you could pass a pageable object like I've used on the get products endpoint but a pageable that only the **size** is taken from the request parameters. page should be always 0, sort criteria will always be name and the direction ascending.